projects/packages/connection/src/class-authorize-json-api.php

```diff
@@ -40,9 +40,17 @@ class Authorize_Json_Api {
 	 * @return void
 	 */
 	public function verify_json_api_authorization_request( $environment = null ) {
-		$environment = $environment === null
-			? $_REQUEST // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- nonce verification handled later in function. We need all of $_REQUEST in order to verify a cryptographic signature of the request data.
-			: $environment;
+		if ( null === $environment ) {
+			$request_params_needed_for_auth = array(
+				'data',
+				'jetpack_json_api_original_query',
+				'nonce',
+				'signature',
+				'timestamp',
+				'token',
+			);
+			$environment                    = array_intersect_key( $request_params_needed_for_auth, $_REQUEST ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended,WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- nonce verification handled later in function and request data are 1) used to verify a cryptographic signature of the request data and 2) sanitized later in function.
```

Comment:
This won't work right, for two reasons:

1. You need to put `$_REQUEST` first so it uses the values from the request.
2. The `$request_params_needed_for_auth` needs to have _keys_ 'data', 'nonce', and so on. So either define it like `'data' => true,` or use `array_flip( $request_params_needed_for_auth )`.

As written, `$environment` will probably be empty. If there's a `$_REQUEST[0]` or the like, it might be `array( 0 => 'data' )` or the like.

Reply:
Good catch, thanks Brad!

Reply:
Thank you Brad and Sergey and apologies for any trouble caused.

I decided to keep things simple and removed the whole logic of extracting the auth request params in `verify_json_api_authorization_request` so we can focus on exactly what the original issue was about.
The debugger related test instructions/unit tests are only applicable for deprecating the input arg in `setup_xmlrpc_handlers` anyways.

I also audited where we actually use `verify_json_api_authorization_request` and it looks like it’s mainly under the context of SSO anyways where the `$environment` is passed [here](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/class.jetpack.php#L935).
The other usage, comes from Jetpack-the-plugin as an action handler for `login_form_jetpack_json_api_authorization` [here](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/class.jetpack.php#L935)

Final PR: https://github.com/Automattic/jetpack/pull/37445